src/server_manager/web_app/app.ts

```diff
@@ -289,319 +312,197 @@ export class App {
-    this.waitForManagedServerCreation();
-  }
-
-  private getDisplayServerBeingCreated(): DisplayServer {
-    if (!this.serverBeingCreated) {
-      return null;
-    }
-    // Set name to the default server name for this region. Because the server
-    // is still being created, the getName REST API will not yet be available.
-    const regionId = this.serverBeingCreated.getHost().getRegionId();
-    const serverName = this.makeLocalizedServerName(regionId);
-    return {
-      // Use the droplet ID until the API URL is available.
-      id: this.serverBeingCreated.getHost().getHostId(),
-      name: serverName,
-      isManaged: true
-    };
+  private addServer(server: server.Server): void {
+    console.log('Loading server', server);
+    const serverId = localId(server);
+    this.idServer.set(serverId, server);
+    const serverEntry = this.makeServerListEntry(server);
+    this.appRoot.serverList = this.appRoot.serverList.concat([serverEntry]);
+
+    // Once the server is added to the list, do the rest asynchronously.
+    setTimeout(async () => {
+      // Wait for server config to load, then update the server view and list.
+      if (isManagedServer(server) && !server.isInstallCompleted()) {
+        try {
+          await (server as server.ManagedServer).waitOnInstall();
+        } catch (error) {
+          if (error instanceof errors.DeletedServerError) {
+            // User clicked "Cancel" on the loading screen.
+            return;
+          }
+          console.log('Server creation failed', error);
+          this.appRoot.showError(this.appRoot.localize('error-server-creation'));
+        }
+      }
+      await this.updateServerView(server);
+      if (this.selectedServer === server) {
+        // Make sure we switch to the server view in case it was in the loading view.
+        this.appRoot.showServerView();
+      }
+      this.appRoot.serverList = this.appRoot.serverList.map(
+          (ds) => ds.id === serverId ? this.makeServerListEntry(server) : ds);
+    }, 0);
   }
 
-  // Shows the last server displayed, if there is one in local storage and it still exists.
-  private async maybeShowLastDisplayedServer() {
-    if (!!this.serverBeingCreated) {
-      // The server being created should be shown regardless of the last user selection.
-      this.displayServerRepository.removeLastDisplayedServerId();
-      return;
-    }
-    const lastDisplayedServerId = this.displayServerRepository.getLastDisplayedServerId();
-    if (!lastDisplayedServerId) {
-      return;  // No server was displayed when user quit the app.
-    }
-    const lastDisplayedServer = this.displayServerRepository.findServer(lastDisplayedServerId);
-    if (!lastDisplayedServer) {
-      return console.debug('Last displayed server ID not found in display sever repository');
+  private removeServer(serverId: string): void {
+    this.idServer.delete(serverId);
+    this.appRoot.serverList = this.appRoot.serverList.filter((ds) => ds.id !== serverId);
```

Comment:
Maybe we should add a `removeServer` (and others) method to `app-root`.

That way `app-root` can be better encapsulated and manage it's own state like `serverList` and `selectedServerId`.

Reply:
The boundary between what's in app-root and what's here is very fuzzy.
We were thinking about the ui components as dumb, without application logic. So this logic really belonged here. But we can consider merging both as an "app".

Reply:
Yeah, that makes sense. I tend to want to have a strongly defined contract with the UI elements, but that may not be idiomatic.